src/Shared/diagnostics.ts

```diff
@@ -207,7 +207,8 @@ export const errors = {
 	),
 
 	// jsx
-	noJsxText: error("JSX text is not supported!"),
+	noPrecedingJsxSpreadElement: error("JSX spread expression must come last in children!"),
+	noJsxFragmentFactory: error("jsxFragmentFactory must be defined to use JSX Fragment expressions!"),
```

Comment:
TS has a default (`React.Fragment`) for this as well, just like the regular jsx factory - is it necessary to have the diagnostic?

Reply:
From my testing, that default doesn't apply if you have a custom jsxFactory. You need to either provide neither or both to use fragments.

Reply:
Right, but as you say, it's neither or both. So I'm not seeing a case where you could have a custom `jsxFactory`, disabling the default, but then not define the fragment factory. I get a TS error with that combination.

Reply:
Decided to hardcode the default for now to match typechecker behavior. If we can find a better way to do this in the future, we can switch.

```ts
	// getJsxFragmentFactoryEntity() doesn't seem to default to "Fragment"..
	// but the typechecker does, so we should follow that behavior
	const jsxFragmentFactoryEntity =
		state.resolver.getJsxFragmentFactoryEntity(node) ??
		ts.parseIsolatedEntityName("Fragment", ts.ScriptTarget.ESNext);
	assert(jsxFragmentFactoryEntity, "Unable to find valid jsxFragmentFactoryEntity");
```